packages/loader/driver-utils/src/mapWithExpiration.ts

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IDisposable } from "@fluidframework/common-definitions";
+
+/**
+ * An extension of Map that expires (deletes) entries after a period of inactivity.
+ * The policy is based on the last time a key was written to.
+ */
+export class MapWithExpiration<TKey, TValue> extends Map<TKey, TValue> implements IDisposable {
+    public disposed: boolean = false;
+    private readonly expirationTimeouts = new Map<TKey, ReturnType<typeof setTimeout>>();
+
+    constructor(
+        private readonly expiryMs: number,
+    ) {
+        super();
+    }
+
+    private scheduleExpiration(key: TKey) {
```

Comment:
I know that timers are relatively expensive, why don't we just store the expiration and then do a basic compare? Doesn't really matter which way you chose, I'm more curious as to why you chose a timer over some time compare.

Reply:
Because it didn't occur to me! I wish I had been pairing with you on this, would have potentially saved a bunch of time, all the Disposable stuff was kind of a pain and may not be necessary

Reply:
Did it.